### PR TITLE
refactor: migrate Divider in typescript

### LINKED ISSUE #896: Divider Component Typescript Migration
TypeScript here we come!
We are migrating to typescript, we are doing so component by component

We would love some help in converting some of our component to typescript - we've created a README
https://github.com/mondaycom/monday-ui-react-core/blob/master/TYPESCRIPT_MIGRATION.md

in this PR we expect you to convert the following files

- [ ] Divider.jsx
- [ ] DividerConstants.js
- [ ] divider.jest.js

Good luck!

### PR DESCRIPTION
**Pull request by aayushbisen**
Convert `Divider` in TypeScript.

Closes #896

#### Basic
- [ ] Used plop (`npm run plop`) to create a new component.
- [x] PR has description.
- [ ] New component is functional and uses Hooks. 
- [ ] Component defines [`PropTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html).
#### Style
- [ ] Styles are added to `NewComponent.modules.scss` file inside of the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [x] Design is compatible with [Monday Design System](https://design.monday.com/).
#### Storybook
- [ ] Stories were added to `/src/NewComponent/__stories__/NewComponent.stories.js` file.
- [x] Stories include all flows of using the component.
#### Tests
- [x] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

### CODE REVIEW

File: src/components/Divider/__stories__/Divider.stories.mdx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import classes from "./Divider.stories.module.scss";
 
 export const metaSettings = createStoryMetaSettings({
   component: Divider,
-  enumPropNamesArray: ["direction"]
+  enumPropNamesArray: ["directions"]
```
Comment (orrgottlieb) [inline src/components/Divider/__stories__/Divider.stories.mdx:9]: why did you change this ?

File: src/components/Divider/Divider.tsx [hunk 2]
```diff
@@ -1,30 +1,25 @@
...
+  classname?: string;
+}
 
 const Divider = ({
   // Backward compatibility for props naming
   classname,
-  className,
-  direction
-}) => {
+  className = undefined,
```
Comment (orrgottlieb) [inline src/components/Divider/Divider.tsx:16]: please leave the `propTypes` for now 🙏

File: src/components/Divider/__tests__/divider.jest.tsx [hunk 3]
```diff
@@ -3,11 +3,11 @@ import renderer from "react-test-renderer";
 import Divider from "../Divider";
 
 it("renders correctly with empty props", () => {
-  const tree = renderer.create(<Divider />).toJSON();
+  const tree = renderer.create(<Divider/>).toJSON();
```
Comment (orrgottlieb) [inline src/components/Divider/__tests__/divider.jest.tsx:6]: prettier fix?

**Review by orrgottlieb — CHANGES_REQUESTED** [review]
Thank you so much. Awesome work! i have a few comments 🙏

File: src/components/Divider/__tests__/divider.jest.tsx (hunk 3 above)
Comment (aayushbisen) [reply src/components/Divider/__tests__/divider.jest.tsx:6]: Yes.

File: src/components/Divider/Divider.tsx (hunk 2 above)
Comment (aayushbisen) [reply src/components/Divider/Divider.tsx:16]: Sure.

**Comment by aayushbisen** [comment]
@orrgottlieb Done the changes for comments.

I noticed that VibeComponentProps has relative path in import in other PR should I do the same?

File: src/components/Divider/__stories__/Divider.stories.mdx (hunk 1 above)
Comment (orrgottlieb) [reply src/components/Divider/__stories__/Divider.stories.mdx]: thanks 🙏

File: src/components/Divider/Divider.tsx (hunk 2 above)
Comment (orrgottlieb) [reply src/components/Divider/Divider.tsx:17]: thank you

**Comment by orrgottlieb** [comment]
@aayushbisen regarding the relative imports - please use relative improts, we need this due to internal usage

**Review by orrgottlieb — APPROVED** [review]
@aayushbisen thank you so much!
Approved, Merged and i will add the correct label to the issue

**Pull request merged**